<launch>
  <arg name="model" default="$(find haptic_device_model)/urdf/haptic_device_model.urdf"/>
  <arg name="rvizconfig" default="$(find haptic_device_model)/rviz/urdf.rviz" />

  <include file="$(find haptic_device_model)/launch/gazebo.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" />

  <rosparam command="load"
            file="$(find haptic_device_model)/config/joints.yaml"
            ns="haptic_joint_state_controller" />
  <rosparam command="load"
            file="$(find haptic_device_model)/config/Jnt1.yaml"
            ns="haptic_Jnt1_controller" />
  <rosparam command="load"
            file="$(find haptic_device_model)/config/Jnt2.yaml"
            ns="haptic_Jnt2_controller" />

  <node name="haptic_controller_spawner" pkg="controller_manager" type="spawner"
    args="haptic_joint_state_controller
          haptic_Jnt1_controller
          haptic_Jnt2_controller
          --shutdown-timeout 3"/>

  <node
    name="joint_state_publisher_gui"
    pkg="joint_state_publisher_gui"
    type="joint_state_publisher_gui" />
          
  <!-- <node name="motionTest" pkg="haptic_device_model" type="test.py" output="screen"/> -->

</launch>
